src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { logActions } from 'app/pages/logs/model/logs.model';
+import { SpinnerService } from 'app/services/spinner.service';
+import { DataResult } from 'app/types/DataResult';
+import { LogAction } from 'app/types/Log';
+import { ServerAction } from 'app/types/Server';
+import { TableColumnDef } from 'app/types/Table';
+import { Observable } from 'rxjs';
+
+import { DialogTableDataSource } from '../dialog-table-data-source';
+
+@Injectable()
+export class LogActionsDialogTableDataSource extends DialogTableDataSource<LogAction> {
+  private reversed = false;
+  constructor(
+    public spinnerService: SpinnerService,
+    public translateService: TranslateService) {
+    super(spinnerService, translateService);
+    // Init
+    this.initDataSource();
+  }
+
+  public loadDataImpl(): Observable<DataResult<LogAction>> {
+    return new Observable((observer) => {
+      const actions: LogAction[] = [];
+      const searchValue = this.getSearchValue().length > 0 ? this.getSearchValue() : '';
```

Comment:
Your statement is equivalent to:
const searchValue = this.getSearchValue();
;-)